core/merginapi.cpp

```diff
@@ -127,8 +127,10 @@ QString MerginApi::listProjectsByName( const QStringList &projectNames )
     return QLatin1String();
   }
 
-  // Authentification is optional in this case, as there might be public projects without the need to be logged in
-  validateAuthAndContinute();
+  // Authentification is optional in this case, as there might be public projects without the need to be logged in.
+  // We only want to include auth token when user is logged in.
+  // User's token, however, might have already expired, so let's just refresh it.
+  refreshAuthToken();
```

Comment:
so what would happen if refresh is not successful? login window will appear?

Reply:
Yes and no, this is public API, client sends a list of projects and server responds with their description. If request is sent without authorization header, private projects will bear only 401/we.